### ISSUE TITLE
Dependency failed for Serial Getty on ttyS0 with Ubuntu Focal Build 

### ISSUE THREAD

**Issue by MMetelko** [RobertCNelson/omap-image-builder#180]
I have been trying to create a ubuntu focal console image and setting up the networking using the netplan approach with systemd-networkd.service.  It looks like the repo is being worked on for Debian Bullseye and Buster, but possibly not tested lately for focal.  I have seen that you are also restructuring some of the rcn-ee/repos for BBB specific setup.  I have looked at where repo changes were made and tried to add patches to be similar to the steps taken for the versions being worked (i.e. adding eth0.network).  I also have a patch that I add for our systems (included below).

 But, at this point, the serial port output stops at failure for Serial Getty on ttyS0 and at the moment I am not able to ssh into the BBB.   I did get an image version to boot (somewhere around 9/21) so that I could ssh into the BBB, but the serial port was still showing this issue.  

A slightly different question, but related ... I have also seen that the repos/bb-usb-gadgets requires the dnsmasq package, but that conflicts with systemd-resolved.service (sharing port 53).  So, I assume I need to disable systemd-resolve.service, but I wonder if the dnsmasq functionality is still needed when using systemd-networkd.service approach.  Since this is in a referenced repo, it seems harder to configure.

 I would appreciate  any suggestions on where to look for the serial port issue and any suggestions on possible conflicting packages for the networking setup.  Let me know if you need any other information.

Using repo commit (9/21): 2d57ac44b83e28ff377bdfebc557ea33c14f9826

Here is the serial output:
======================
U-Boot SPL 2019.04-00002-gbb4af0f50f (Jul 08 2019 - 11:44:39 -0500)
Trying to boot from MMC2
Loading Environment from EXT4... ** File not found /boot/uboot.env **

** Unable to read "/boot/uboot.env" from mmc0:1 **

U-Boot 2019.04-00002-gbb4af0f50f (Jul 08 2019 - 11:44:39 -0500), Build: jenkins-github_Bootloader-Builder-128

CPU  : AM335X-GP rev 2.1
I2C:   ready
DRAM:  512 MiB
No match for driver 'omap_hsmmc'
No match for driver 'omap_hsmmc'
Some drivers were not found
Reset Source: Power-on reset has occurred.
RTC 32KCLK Source: External.
MMC:   OMAP SD/MMC: 0, OMAP SD/MMC: 1
Loading Environment from EXT4... ** File not found /boot/uboot.env **

** Unable to read "/boot/uboot.env" from mmc0:1 **
Board: BeagleBone Black
<ethaddr> not set. Validating first E-fuse MAC
BeagleBone Black:
BeagleBone: cape eeprom: i2c_probe: 0x54:
BeagleBone: cape eeprom: i2c_probe: 0x55:
BeagleBone: cape eeprom: i2c_probe: 0x56:
BeagleBone: cape eeprom: i2c_probe: 0x57:
Net:   eth0: MII MODE
cpsw, usb_ether
Press SPACE to abort autoboot in 2 seconds
board_name=[A335BNLT] ...
board_rev=[00C0] ...
switch to partitions #0, OK
mmc0 is current device
SD/MMC found on device 0
gpio: pin 56 (gpio 56) value is 0
gpio: pin 55 (gpio 55) value is 0
gpio: pin 54 (gpio 54) value is 0
gpio: pin 53 (gpio 53) value is 1
starting USB...
USB0:   Port not available.
gpio: pin 54 (gpio 54) value is 1
Checking for: /boot/uEnv.txt ...
switch to partitions #0, OK
mmc0 is current device
gpio: pin 56 (gpio 56) value is 0
gpio: pin 55 (gpio 55) value is 0
gpio: pin 54 (gpio 54) value is 0
gpio: pin 53 (gpio 53) value is 1
switch to partitions #0, OK
mmc0 is current device
gpio: pin 54 (gpio 54) value is 1
Checking for: /uEnv.txt ...
Checking for: /boot.scr ...
Checking for: /boot/boot.scr ...
Checking for: /boot/uEnv.txt ...
gpio: pin 55 (gpio 55) value is 1
2138 bytes read in 39 ms (52.7 KiB/s)
Loaded environment from /boot/uEnv.txt
Checking if uname_r is set in /boot/uEnv.txt...
gpio: pin 56 (gpio 56) value is 1
Running uname_boot ...
loading /boot/vmlinuz-5.4.106-ti-rt-r35 ...
10306048 bytes read in 677 ms (14.5 MiB/s)
debug: [enable_uboot_overlays=1] ...
debug: [enable_uboot_cape_universal=] ...
debug: [uboot_base_dtb_univ=am335x-boneblack-uboot.dtb] ...
uboot_overlays: [uboot_base_dtb=am335x-boneblack-uboot.dtb] ...
uboot_overlays: Switching too: dtb=am335x-boneblack-uboot.dtb ...
loading /boot/dtbs/5.4.106-ti-rt-r35/am335x-boneblack-uboot.dtb ...
92818 bytes read in 89 ms (1017.6 KiB/s)
uboot_overlays: [fdt_buffer=0x60000] ...
uboot_overlays: loading /lib/firmware/BB-BONE-eMMC1-01-00A0.dtbo ...
1614 bytes read in 821 ms (1000 Bytes/s)
uboot_overlays: loading /lib/firmware/BB-HDMI-TDA998x-00A0.dtbo ...
5330 bytes read in 385 ms (12.7 KiB/s)
uboot_overlays: loading /lib/firmware/BB-ADC-00A0.dtbo ...
654 bytes read in 2108 ms (0 Bytes/s)
uboot_overlays: unable to find [mmc 0:1 AM335X-PRU-UIO-00A0.dtbo]...
uboot_overlays: add [enable_uboot_cape_universal=1] to /boot/uEnv.txt to enable...
loading /boot/initrd.img-5.4.106-ti-rt-r35 ...
9112944 bytes read in 599 ms (14.5 MiB/s)
debug: [console=ttyS0,115200n8 root=/dev/mmcblk0p1 ro rootfstype=ext4 rootwait coherent_pool=1M net.ifnames=0 security=apparmor lpj=1990656 rng_core.default_quality=100 quiet] ...
debug: [bootz 0x82000000 0x88080000:8b0d70 88000000] ...
## Flattened Device Tree blob at 88000000
   Booting using the fdt blob at 0x88000000
   Loading Ramdisk to 8f74f000, end 8ffffd70 ... OK
   Loading Device Tree to 8f6d4000, end 8f74efff ... OK

Starting kernel ...

[    0.001169] 000: timer_probe: no matching timers found
[    0.108059] 000: l3-aon-clkctrl:0000:0: failed to disable
[    0.189394] 000: debugfs: Directory '49000000.edma' with parent 'dmaengine' already present!
[    1.636579] 000: wkup_m3_ipc 44e11324.wkup_m3_ipc: could not get rproc handle
[    1.711789] 000: omap_hsmmc 47810000.mmc: RX DMA channel request failed
[    2.011784] 000: davinci-mcasp 48038000.mcasp: IRQ common not found
[    2.085318] 000: omap_voltage_late_init: Voltage driver support not added
[    2.085772] 000: PM: Cannot get wkup_m3_ipc handle
[.[0;1;31mFAILED.[0m] Failed to start .[0;1;39mEnable File System Quotas.[0m.
[.[0;1;31m TIME .[0m] Timed out waiting for device .[0;1;39m/dev/ttyS0.[0m.
[.[0;1;38;5;185mDEPEND.[0m] Dependency failed for .[0;1;39mSerial Getty on ttyS0.[0m.

The patch I am using is:
======================
diff --git a/scripts/chroot.sh b/scripts/chroot.sh
index a96b6e61..711c9432 100755
--- a/scripts/chroot.sh
+++ b/scripts/chroot.sh
@@ -444,6 +444,11 @@ bookworm|sid)
 	echo "##deb-src http://deb.debian.org/debian ${deb_codename}-updates ${deb_components}" >> ${wfile}
 	echo "" >> ${wfile}
 	;;
+focal)
+	echo "deb http://${deb_mirror} ${deb_codename}-updates ${deb_components}" >> ${wfile}
+	echo "#deb-src http://${deb_mirror} ${deb_codename}-updates ${deb_components}" >> ${wfile}
+	echo "" >> ${wfile}
+	;;
 esac
 
 #https://wiki.debian.org/Backports
@@ -1100,9 +1105,19 @@ cat > "${DIR}/chroot_script.sh" <<-__EOF__
 		fi
 
 		if [ ! "x${rfs_use_systemdnetworkd}" = "x" ] ; then
-			if [ -f /etc/bbb.io/templates/eth0-DHCP.network ] ; then
-				cp -v /etc/bbb.io/templates/eth0-DHCP.network /etc/systemd/network/eth0.network
-			fi
+			echo "[Match]" > /etc/systemd/network/eth0.network
+			echo "Name=eth0" >> /etc/systemd/network/eth0.network
+			echo "Type=ether" >> /etc/systemd/network/eth0.network
+			echo "" >>  /etc/systemd/network/eth0.network
+			echo "[Link]" > /etc/systemd/network/eth0.network
+			echo "RequiredForOnline=no" >> /etc/systemd/network/eth0.network
+			echo "" >>  /etc/systemd/network/eth0.network
+			echo "[Network]" >>  /etc/systemd/network/eth0.network
+			echo "DHCP=ipv4" >>  /etc/systemd/network/eth0.network
+
+			#if [ -f /lib/systemd/system/systemd-resolved.service ] ; then
+			#	systemctl disable systemd-resolved.service || true
+			#fi
 
 			if [ -f /lib/systemd/system/systemd-networkd.service ] ; then
 				systemctl enable systemd-networkd.service || true
@@ -1523,7 +1538,8 @@ if [ -f "${tempdir}/etc/dogtag" ] ; then
 fi
 
 #Add Google IPv4 nameservers
-if [ -f "${tempdir}/etc/resolv.conf" ] ; then
+if [ ! -f "${tempdir}/etc/resolv.conf" ] ; then
+	sudo rm ${tempdir}/etc/resolv.conf || true
 	wfile="${tempdir}/etc/resolv.conf"
 	sudo sh -c "echo 'nameserver 8.8.8.8' > ${wfile}"
 	sudo sh -c "echo 'nameserver 8.8.4.4' >> ${wfile}"
diff --git a/tools/setup_sdcard.sh b/tools/setup_sdcard.sh
index ce2724c1..0e0f3ebf 100755
--- a/tools/setup_sdcard.sh
+++ b/tools/setup_sdcard.sh
@@ -1227,7 +1227,7 @@ populate_rootfs () {
 			echo "console=${extlinux_console}" >> ${wfile}
 		fi
 
-		cmdline="coherent_pool=1M net.ifnames=0"
+		cmdline="coherent_pool=1M net.ifnames=0 security=apparmor"
 
 		if [ ! "x${loops_per_jiffy}" = "x" ] ; then
 			cmdline="${cmdline} ${loops_per_jiffy}"
@@ -1371,7 +1371,7 @@ populate_rootfs () {
 		if [ "${BTRFS_FSTAB}" ] ; then
 			echo "${rootfs_drive}  /  btrfs  defaults,noatime  0  1" >> ${wfile}
 		else
-			echo "${rootfs_drive}  /  ${ROOTFS_TYPE}  noatime,errors=remount-ro  0  1" >> ${wfile}
+			echo "${rootfs_drive}  /  ${ROOTFS_TYPE}  noatime,errors=remount-ro,usrquota,grpquota  0  1" >> ${wfile}
 		fi
 
 		if [ "x${uboot_efi_mode}" = "xenable" ] ; then

omap-image

Key entries in the my-focal.conf used in the build (items not related to this question removed to shorten listing):
================================================================================##
release="20.04.2"
image_type="console"
##
##Debootstrap: https://wiki.debian.org/Debootstrap
##
##debootstrap --arch=${deb_arch} --include=${deb_include} --exclude=${deb_exclude} \
##--components=${deb_components} --foreign ${deb_codename} /tmp/tmp.dir/ http://${deb_mirror}
##
deb_distribution="ubuntu"
deb_codename="focal"
deb_arch="armhf"
#
# includes in RobertCNelson version, but not used here:  cloud-guest-utils, alsa-utils, avahi-utils, bluetooth, connman, dnsmasq
deb_include="	\
	acpid	\
	bash-completion	\
	bc	\
	bsdmainutils	\
	build-essential	\
	ca-certificates	\
	can-utils	\
	cpufrequtils	\
	crda	\
	device-tree-compiler	\
	dosfstools	\
	fake-hwclock	\
	fbset	\
	file	\
	git	\
	gnupg	\
	haveged	\
	hexedit	\
	hostapd	\
	htop	\
	i2c-tools	\
	initramfs-tools	\
	iw	\
	linux-base	\
	linux-firmware	\
	locales	\
	lsb-release	\
	make	\
	nano	\
	nethogs	\
	net-tools	\
	nginx	\
	openssh-server	\
	pastebinit	\
	patch	\
	pkg-config	\
	ppp	\
	pps-tools	\
	read-edid	\
	rfkill	\
	rsync	\
	sudo	\
	systemd	\
	systemd-sysv	\
	tio	\
	u-boot-tools	\
	usb-modeswitch	\
	usbutils	\
	wget	\
	wireless-regdb	\
	wireless-tools	\
	wpasupplicant	\
"

#
deb_exclude=""
#
deb_components="main universe multiverse"
deb_mirror=""
##
##Some packages fail to install via debootstrap: deb_additional_pkgs="<comma|space>"
##
deb_additional_pkgs="	\
	gpsd	\
	grub-efi-arm	\
	ifupdown	\
	iptables	\
	libnss-mdns	\
	libnss-systemd	\
	libpam-systemd	\
	libsystemd-dev	\
	net-tools	\
	quota	\
"

##
rfs_default_locale="en_US.UTF-8"
rfs_etc_dogtag="rcn-ee.net Ubuntu 20.04.2 Console Image"
rfs_console_banner="Support: http://elinux.org/BeagleBoardUbuntu"
rfs_console_user_pass="enable"
rfs_ssh_banner="Support: http://elinux.org/BeagleBoardUbuntu"
rfs_ssh_user_pass="enable"
rfs_use_systemdnetworkd="enable"
##
##enable kernel repo: http://repos.rcn-ee.com/(debian|ubuntu)
##Beaglebone Black uses am335x, while Beagleboard-x15 and Beaglebone-AI use am57xx
##configuring for Beaglebone Black only at this time, so removing bb-u-boot-am57xx-evm
repo_rcnee="enable"
repo_rcnee_arch="armhf"
repo_rcnee_pkg_list="	\
	bb-bbai-firmware	\
	bb-cape-overlays	\
	bb-customizations	\
	bb-usb-gadgets	\
	bb-u-boot-am335x-evm	\
	bb-wl18xx-firmware	\
	gpiod	\
	ipumm-dra7xx-installer	\
	linux-image-5.4.106-ti-rt-r35	\
	moreutils
	vpdma-dra7xx-installer	\
	xauth	\
"

repo_rcnee_pkg_version="5.4.106-ti-rt-r35"
repo_rcnee_cmem_version="4.16.00.00"
include_firmware="enable"

**Comment by RobertCNelson** [comment]
@MMetelko yeah sorry!  I'm pretty happy with how Bullseye is being configured now, i'll be working on backporting this to Buster and Ubuntu LTS..

correct, systemd-network removes our need for dnsmasq!!! 

the udev rules for wifi really cleanup the out of box tether stack..

We just need gadget support in systemd:

```
voodoo@x6-am57xx-beagle-x15-2gb:/lib/systemd$ find . | grep gadget
./system/usb-gadget.target
```

**Comment by MMetelko** [comment]
@RobertCNelson 
Given your constant activity in updating this and the rcn-ee/repos/, I can tell you probably have a substantial list of priorities you are working through daily.  Is it possible to provide a possible time frame when you might get to a stable point where backporting makes sense?  I do understand that things happen and desired deadlines can even get delayed, so I am looking for a general timeframe so that I can set expectations on the project I am using this repo to create BBB images.  

It is far down the road, can you help me understand where to find the setup for the `usb-gadget.target`?  Doing the find does show that it is not setup in the builds I have created.

**Comment by RobertCNelson** [comment]
Nah, wpa is still blocking... as long as you have a wifi adapter.. it's fast..

**Comment by MMetelko** [comment]
I am still missing something.  Here is what I am running after ./RootStock-NG:

```
IMG=$(\ls deploy)
cd deploy/$IMG
sudo ./setup_sdcard.sh --img-4gb $IMG --dtb beaglebone --distro-bootloader
```

Here is what I am seeing:
```
Log: copying setup_sdcard.sh related files
Log: packaging rootfs: [armhf-rootfs-ubuntu-focal.tar]
-rw-r--r-- 1 root root 964M Oct  8 22:53 /var/lib/jenkins/workspace/ld_dev-5.4-rt-kernel-image-build/omap-image-builder/deploy/ubuntu-20.04.3-console-armhf-2021-10-08/armhf-rootfs-ubuntu-focal.tar
Log: USER:jenkins
syncing media... 
0+0 records in
0+0 records out
0 bytes copied, 0.000293332 s, 0.0 kB/s
-----------------------------
U-Boot Overlays Enabled...
Debug: ARM rootfs: armhf-rootfs-ubuntu-focal.tar

Using Distro Bootloader
-----------------------------
cp: cannot stat './u-boot/bb-u-boot-am335x-evm/MLO': No such file or directory
```

**Comment by MMetelko** [comment]
Yes, I followed the pattern of rcn-ee.net-console-ubuntu-focal-v5.10-ti-armhf.conf since I want to use the netplan systemd approach to networking.  

Here are the packages I included in the `repo_rcnee_pkg_list`:
```
	bb-bbai-firmware	\
	bb-beagle-config	\
	bb-beagle-version	\
	bb-cape-overlays	\
	bb-usb-gadgets	\
	bb-u-boot-am335x-evm	\
	bb-wl18xx-firmware	\
	bb-wlan0-defaults	\
	generic-sys-mods	\
	gt	\
	libusbgx	\
	linux-image-5.4.106-ti-rt-r35	\
	overlayroot	\
```

I also included `u-boot-tools` in the `deb_include` list
